File: PADTiltViewController/PADTiltTableViewController.m
//
//  PADTiltTableViewController.m
//  PADTiltViewController
//
//  Created by Pavel Dusatko on 5/3/14.
//  Copyright (c) 2014 Pavel Dusatko. All rights reserved.
//

#import "PADTiltTableViewController.h"
#import "UIViewController+PADTiltAdditions.h"

@implementation PADTiltTableViewController

#pragma mark - NSObject - Creating, Copying, and Deallocating Objects

- (void)dealloc {
    [self pad_dealloc];
}

- (instancetype)init {
    self = [super init];
    if (self) {
        [self pad_init];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self pad_init];
    }
    return self;
}

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        [self pad_init];
    }
    return self;
}

- (instancetype)initWithStyle:(UITableViewStyle)style {
    self = [super initWithStyle:style];
    if (self) {
        [self pad_init];
    }
    return self;
}

#pragma mark - UIViewController - Responding to View Events

- (void)viewDidDisappear:(BOOL)animated {
    [super viewDidDisappear:animated];
    
    [self pad_stopReceivingTiltUpdates];
}

#pragma mark - UIViewController - Responding to View Rotation Events

- (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration {
    [super willRotateToInterfaceOrientation:toInterfaceOrientation duration:duration];
    
    [self pad_stopReceivingTiltUpdates];
}

@end

File: Swift-version/Inpiration/PADTiltTableViewController.swift
//
//  PADTiltTableViewController.swift
//  Inpiration
//
//  Created by Alexander Hüllmandel on 10/31/14.
//  Copyright (c) 2014 Alexander Hüllmandel. All rights reserved.
//

import Foundation
import UIKit

/**
*  @abstract A table view controller that adds device tilt capability to one directional table views.
*/
class PADTiltTableViewController: UITableViewController {
    //MARK: - NSObject - Creating, Copying, and Deallocating Objects
    
    deinit {
        pad_dealloc()
    }
    
    override init() {
        super.init()
        pad_init()
    }
    
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        pad_init()
    }
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        pad_init()
    }
    
    override init(style: UITableViewStyle) {
        super.init(style: style)
        pad_init()
    }
    
    //MARK: - UIViewController - Responding to View Events
    override func viewDidDisappear(animated:Bool) {
        super.viewDidDisappear(animated)
        pad_stopReceivingTiltUpdates()
    }
    
    //MARK: - UIViewController - Responding to View Rotation Events
    override func willRotateToInterfaceOrientation(toInterfaceOrientation: UIInterfaceOrientation, duration: NSTimeInterval) {
        super.willRotateToInterfaceOrientation(toInterfaceOrientation, duration: duration)
        pad_stopReceivingTiltUpdates()
    }
}